### PR TITLE
[serviceconnector-passwordless] add extension information in user agent

### PR DESCRIPTION
**Pull request by xfz11**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Review by yungezz — APPROVED** [review]

**Comment by yonzhan** [comment]
@kairu-ms for awareness

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+
+# pylint: disable=raise-missing-from, consider-using-f-string
+def _is_vendored_sdk_path(path_comps):
+    return len(path_comps) >= 5 and path_comps[4] == 'vendored_sdks'
+
+
+def resolve_client_arg_name(operation, kwargs):
+    if not isinstance(operation, str):
+        raise CLIError("operation should be type 'str'. Got '{}'".format(type(operation)))
```
Comment (kairu-ms) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py:25]: Please use other error types. `CLIError` is not recommended to use.

Comment (xfz11) [reply src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py:25]: Thanks for reminding. This file is just the copy of `client_factory.py` in core. We don't want to change the original implementation and just need to add extension name and version in user agent by adding 2 more lines.

File: src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    # Prepare connection_verify to change SSL verification behavior, used by ConnectionConfiguration
+    client_kwargs.update(_debug.change_ssl_cert_verification_track2())
+
+    # Prepare User-Agent header, used by UserAgentPolicy
+    client_kwargs['user_agent'] = get_az_user_agent()
+    try:
+        command_ext_name = cli_ctx.data['command_extension_name']
+        command_ext_version = cli_ctx.data['command_extension_version']
+        if command_ext_name:
+            client_kwargs['user_agent'] += " CliExtension/{}({})".format(command_ext_name, command_ext_version)
```
Comment (xfz11) [inline src/serviceconnector-passwordless/azext_serviceconnector_passwordless/client_factory_util.py:147]: This is the added line with extension version

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**